### DISCUSSION TITLE
Is there a way to interpolate variables into .testrailrc for browser/OS?

### DISCUSSION THREAD

**Discussion by ebarila** [DamianOsipiuk/testcafe-reporter-testrail#21] [Q&A]
I'm trying to add the test result tittle on Testrail appending to it the Broser type and operative system. But it doesn't seem to take variables. Is there a way to do it?

**Comment by DamianOsipiuk** [comment]
Hi, You can use `TESTRAIL_RUN_NAME` env variable to set it up.
The order of precedence is:
1. ENV variables
2. Config file
3. Defaults

However you might want to think about running tests on multiple browsers for the same test run. It will then contain results for all of the browsers under the same entry. Testcafe supports this out of the box.
I believe when I have last checked Testrail also handled that well.

**Reply by ebarila** [reply]
Thanks for the answer, but could you provide an example on how to set up that env variable? I have not set up this integration and its new to me. 
Glad to know you can do multiple browsers at once, but seeing how long each browser takes, we will keep doing once at the time.

**Reply by DamianOsipiuk** [reply]
It depends on how the integration is done, and how do you invoke the tests.
But trying to make it as easy as possible, if you run tests by just invoking `testcafe`, it would be something like
```
TESTRAIL_RUN_NAME='%BRANCH%#%BUILD% - %DATE% (Chrome, Linux)' testcafe
```
This is just an example of adding `(Chrome, Linux)` to the default name that is generated. This example should work for both Mac and Linux, but Windows has slightly different syntax.

If you invoke your tests from javascript, then you can set the env variable via `process.env.TESTRAIL_RUN_NAME='...'`
